packages/cactus-plugin-ledger-connector-xdai/src/main/typescript/plugin-ledger-connector-xdai.ts

```diff
@@ -322,6 +297,100 @@ export class PluginLedgerConnectorXdai
     }
   }
 
+  public async invokeContract(
+    req: InvokeContractV1Request,
+  ): Promise<InvokeContractV1Response> {
+    const fnTag = `${this.className}#invokeContract()`;
+    const { contractName, keychainId } = req;
+    if (!contractName) {
+      throw new Error(
+        `${fnTag} Cannot recover the keychain plugin because the contractName is empty`,
+      );
+    }
+    if (!keychainId) {
+      throw new Error(`${fnTag} Cannot invoke contract without keychainId`);
+    }
+    const keychainPlugin = this.pluginRegistry.findOneByKeychainId(keychainId);
+    if (!keychainPlugin.has(contractName)) {
+      throw new Error(
+        `${fnTag} Cannot invoke the contract because contractName is not in the keychainPlugin`,
+      );
+    }
+    const contractStr = await keychainPlugin.get(contractName);
+    const contractJSON = JSON.parse(contractStr);
+    (req as any).contractJSON = contractJSON; // remove
```

Comment:
@elenaizaguirre This line can be removed now, right?

Reply:
@petermetz Sure, sorry for the confusion